### ISSUE TITLE
MD022 per-level array config (`lines-above = [...]`) is documented and unit-tested, but rejected at runtime as a type mismatch

### ISSUE THREAD

**Issue by andreashaerter** [rvben/rumdl#841]
### rumdl version

0.2.62

### Installation method

Other

### Bug description

Frist of all: Installation method == `uv` (cannot chose this) but also reproduced with the binary download; I would be really sorry if false alarm, but I *really* checked.

https://rumdl.dev/md022/#per-level-configuration documents a per-heading-level array form for `MD022.lines-above` and `MD022.lines-below`, for example `lines-above = [2, 1, 1, 1, 1, 1]` for h1 through h6, including a `-1` sentinel to inherit the default for a given level. Exactly what I wished for (and other formatters don't provide; `rumdl` is really flexible).

However: This syntax is rejected at runtime with `[config warning] Type mismatch for MD022.lines-above: expected integer, got array`, and the rule silently falls back to its scalar default instead.

This isn't a case of the docs describing an unreleased feature. I checked the actual `v0.2.62` git tag: the commit that added this (`64b75e36`, "feat(md022): add per-level unlimited blank handling") is an ancestor of the `v0.2.62` tag, and the tagged source tree contains a full `HeadingLevelConfig::PerLevel` implementation with its own unit test, `test_per_level_array_configuration`, which directly asserts that `lines-above = [0, 1, 1, 2, 2, 2]` parses into per-level values. So the released source has this implemented and passing its own tests, but whatever code path actually loads `--config`/`.rumdl.toml` values into the rule at runtime does not appear to go through it.

I searched existing issues and found no exact match. The two closest (#429, #449) are older MD022 config bugs from the 0.1.x era, already fixed, and about different behavior (a `MD012` interaction, and a "more than one blank line not enforced" bug).

### Steps to reproduce

1. Create `t.md`:

   ```markdown
   # Title

   ## Section

   ### Sub

   Content.
   ```

2. Run, using the docs' own example values:

   ```sh
   rumdl check --no-config --deny-config-warnings \
     --config 'MD022.lines-above=[2,1,1,1,1,1]' t.md
   ```

3. Observe the config warning instead of per-level spacing being applied.

Also reproduces via a real config file:

```toml
[MD022]
lines-above = [2, 1, 1, 1, 1, 1]
lines-below = [1, 1, 1, 0, 0, 0]
```

And with the documented `-1` inherit-default form:

```toml
[MD022]
lines-above = [-1, 2, 1, 1, 1, 1]
```

Same warning in all three forms.

<img width="793" height="773" alt="Image" src="https://github.com/user-attachments/assets/dc00cbcf-b54a-464a-803f-4bc2350d1a27" />

### Expected behavior

The array should parse as documented and as the shipped unit test asserts: each heading level gets its own required blank-line count, with `-1` inheriting that level's default.

### Actual behavior

```
[config warning] Type mismatch for MD022.lines-above: expected integer, got array
```

The rule falls back to its default (scalar) behavior. `--deny-config-warnings` turns this into a hard error (exit 2), so a CI pipeline relying on that flag would fail outright rather than silently ignoring the array.

### Configuration

```toml

```

### Operating system

Linux

### Additional context

I cross-verified this across three independent installation and distribution channels, specifically to rule out a stale or mismatched package build before reporting:

1. `uvx --from rumdl rumdl` (PyPI, via `uv tool`)
2. A clean `uv venv` plus `uv pip install rumdl` (PyPI wheel, installed directly, no `uvx` caching involved)
3. The raw prebuilt binary from GitHub Releases (`rumdl-v0.2.62-x86_64-unknown-linux-gnu.tar.gz`, downloaded and run directly, no Python or PyPI involved at all)

All three report `rumdl 0.2.62` via `--version` and produce the exact same type-mismatch warning on the exact same input. That rules out a packaging/distribution mismatch, such as a PyPI wheel not rebuilt from the
tagged source, as the explanation.

I did not trace further into the Rust source to find exactly where the disconnect is, since I'm not familiar with the codebase's internals, but given the unit test passes in isolation, my guess is that the rule's actual
runtime config struct (whatever `--config`/TOML deserialization constructs and hands to the rule) still has a plain `usize`/scalar field for `lines-above`/`lines-below` rather than routing through `HeadingLevelConfig`, so the two never got connected end to end.

This seems to be *not* a regression. I checked `v0.0.180`, the first tagged release that includes `64b75e36`, using a discriminating test case: a document where an h2 has only one blank line above it, which `lines-above = [3, 3, 1, 1, 1, 1]` should flag (needs three) but the scalar default of one would not. `v0.0.180` reports no issues, meaning the array is silently ignored there too, just without the explicit config-warning message (that appears to have been added later; `v0.0.180` also predates `--deny-config-warnings`, which doesn't exist as a flag in that version). So this might have never worked in any released version.

**Comment by rvben** [comment]
Thanks @andreashaerter for the detailed report and cross-version verification. The diagnosis was right: MD022’s deserializer already accepted either an integer or a six-entry array, but configuration validation inferred the integer type from the scalar defaults and rejected the documented array form first.

Fixed by `976087c5`. Both `lines-above` and `lines-below` are now recognized as polymorphic configuration keys. I also added end-to-end coverage for inline configuration and a TOML configuration file, including `-1`, with `--deny-config-warnings`; the tests verify that the per-heading-level values affect MD022 behavior.

Closing as fixed. This will be included in the next release. Thanks again for the precise reproducer.
